src/storage/src/hummock/store/version.rs

```diff
@@ -644,22 +686,11 @@ impl HummockVersionReader {
                     Arc::new(SstableIteratorReadOptions::default()),
                 ));
             } else {
-                let table_infos = prune_ssts(
-                    level.table_infos.iter(),
-                    read_options.table_id,
-                    &table_key_range,
-                );
-                if table_infos.is_empty() {
-                    continue;
-                }
-                // Overlapping
                 let mut iters = Vec::new();
-                for table_info in table_infos.into_iter().rev() {
-                    let sstable = self
-                        .sstable_store
-                        .sstable(table_info, &mut local_stats)
-                        .in_span(Span::enter_with_local_parent("get_sstable"))
-                        .await?;
+                for _sstable_info in fetch_meta_req {
+                    let (sstable, local_cache_meta_block_miss) =
+                        flatten_resps.pop().unwrap().unwrap();
+                    local_stats.apply_meta_fetch(local_cache_meta_block_miss);
```

Comment:
Can we add a assertion to make sure the sstable matches?
```
assert_eq!(sstable_info.id, sstable.value().id);
```

Reply:
> Can we add a assertion to make sure the sstable matches?
> 
> ```
> assert_eq!(sstable_info.id, sstable.value().id);
> ```

fixed